Add achievement types documentation (#192)
Gamify-IT/docs: dev-manuals/achievements/achievement-types.md

Typical achievements would be something like this: `Kill x enemies`, `Reach 1st place on the leaderboard`, ...  
For Gamify-IT, achievements like `Reach 1st place on the leaderboard` or `Win x rounds of Towercrush` are possible.

### Achiever

The achiever wants to progress as fast as possible and doesn't really care about unnecessary side quests.  
Typical achievements would be something like this: `Reach level x`, `Complete chapter x`, ...  
For Gamify-IT, achievements like `Complete world x` or `Unlock all areas` are possible.

### Explorer

The explorer wants to explore the world and find hidden areas and items.  
Typical achievements would be something like this: `Find x hidden objects`, ...  
For Gamify-IT, achievements like `Find all easter eggs`, `Travel a distance of x tiles` or `Activate all teleporters` are possible.

### Socializer

The socializer wants to interact and work together with other players.  
Typical achievements would be something like this: `Add x friends`, `Player x rounds of multiplayer`, ...  
For Gamify-IT, achievements like `Player x rounds of Towercrush` or `Player x times with the same player` are possible.

## Achievement goals

> Completely (not shown at all until unlocked) or partially hidden (only title visible until unlocked) achievements can be an additional motivation to play. This is especially true for explorers and achievers.  

Reviewer: Also, I'm still missing whether the achievements will be made globally for **every** course, or if every course can have its own achievements.
My personal preference would be achievements per course.

But every player has all player types at his own personal weighting, that means that all achievements are relevant for all players, just at different importance.  

> It is just important that the achievement has a purpose. A player has to get it either by progressing the game or by doing something special. There should not be any achievements earn for simply investing an absurd amount of time, for example `Player 10.000 minigames`, `Reach 1.000 times a result of 100%` or `Travel a distance of 100.000.000 tiles`.

Reviewer: ```suggestion
It is important that the achievement has a purpose.  
A player has to get it either by progressing in the game or by doing something special.  
No achievement should be earned simply by investing an absurd amount of time such as `Play 10.000 minigames`, `Reach 1.000 times a result of 100%` or `Travel a distance of 100.000.000 tiles`.
```
